src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/HttpGoogleCloudStorageDAO.scala

```diff
@@ -241,5 +258,11 @@ class HttpGoogleCloudStorageDAO(
     s"${groupsPrefix}-${workspaceName.namespace}_${workspaceName.name}-${WorkspaceAccessLevel.toCanonicalString(accessLevel)}@${appsDomain}"
   }
 
+  def deconstructGroupId(groupId: String): (WorkspaceName, WorkspaceAccessLevel) = {
+    val strippedId = groupId.stripPrefix(s"${groupsPrefix}-").stripSuffix(s"@${appsDomain}").split("_", 2)
+    (WorkspaceName(strippedId.head, strippedId(1).splitAt(strippedId(1).lastIndexOf("-"))._1),
+      WorkspaceAccessLevel.fromCanonicalString(strippedId(1).splitAt(strippedId(1).lastIndexOf("-"))._2.stripPrefix("-").toUpperCase))
```

Comment:
We should probably add tests for this method to make sure that it plays well with underscores and dashes in workspace name / namespace (both now and in the future).